android/src/org/ppsspp/ppsspp/TextRenderer.java

```diff
@@ -0,0 +1,47 @@
+package org.ppsspp.ppsspp;
+import android.graphics.*;
+
+import java.nio.ByteBuffer;
+
+public class TextRenderer {
+	private static Paint p;
+	private static Paint bg;
+	static {
+		p = new Paint(Paint.SUBPIXEL_TEXT_FLAG | Paint.ANTI_ALIAS_FLAG);
+		p.setColor(Color.WHITE);
+		bg = new Paint();
+		bg.setColor(Color.BLACK);
+	}
+	public static int measureText(String string, double textSize) {
+		Rect bound = new Rect();
+		p.setTextSize((float)textSize);
+		p.getTextBounds(string, 0, string.length(), bound);
+		int w = bound.width();
+		int h = (int)(p.descent() - p.ascent() + 2.0f);
+		// Round width up to even already here to avoid annoyances from odd-width 16-bit textures which
+		// OpenGL does not like - each line must be 4-byte aligned
+		w = (w + 3) & ~1;
+		return (w << 16) | h;
+	}
+	public static int[] renderText(String string, double textSize) {
+		Rect bound = new Rect();
+		p.setTextSize((float)textSize);
+		p.getTextBounds(string, 0, string.length(), bound);
+		int w = bound.width();
+		int h = (int)(p.descent() - p.ascent() + 2.0f);
+		// Round width up to even already here to avoid annoyances from odd-width 16-bit textures which
+		// OpenGL does not like - each line must be 4-byte aligned
+		w = (w + 3) & ~1;
+
+		float baseline = -p.ascent();
+		Bitmap bmp = Bitmap.createBitmap(w, h, Bitmap.Config.ARGB_8888);
+		Canvas canvas = new Canvas(bmp);
+		canvas.drawRect(0.0f, 0.0f, w, h, bg);
+		p.setColor(Color.WHITE);
+		canvas.drawText(string, 1, -p.ascent() + 1, p);
```

Comment:
`baseline + 1`?  Or should we remove `baseline`?

Also - why origin at (1, 1) anyway?  Is it clipping out aliasing otherwise?

-[Unknown]